### PR TITLE
More fuzz tests for common http utility

### PR DESCRIPTION
**Pull request by ericlin0101**
Signed-off-by: Lin Lin <linlineric@google.com>

For an explanation of how to fill out the fields, please see the relevant section
in [PULL_REQUESTS.md](https://github.com/envoyproxy/envoy/blob/master/PULL_REQUESTS.md)

Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
[Optional Fixes #Issue]
[Optional Deprecated:]

### CODE REVIEW

File: test/common/http/utility_fuzz_test.cc [hunk 1]
```diff
@@ -5,6 +5,8 @@
 #include "test/fuzz/utility.h"
 #include "test/test_common/utility.h"
 
+//#include "common/http/header_map_impl.h"
```
Comment (asraa) [inline test/common/http/utility_fuzz_test.cc]: nit: remove comment

File: test/common/http/utility_fuzz_test.cc [hunk 2]
```diff
@@ -41,6 +41,29 @@ DEFINE_PROTO_FUZZER(const test::common::http::UtilityTestCase& input) {
     Http::Utility::extractHostPathFromUri(input.extract_host_path_from_uri(), host, path);
     break;
   }
+  case test::common::http::UtilityTestCase::kPercentEncodingString: {
```
Comment (asraa) [inline test/common/http/utility_fuzz_test.cc]: Is there any idempotent property you can test here? i.e. decoding the encoded giving you the original

Comment (asraa) [reply test/common/http/utility_fuzz_test.cc]: I actually don't think there is.

**Comment by ericlin0101** [comment]
Done, the FUZZ_ASSERT in percent encoding.

**Pull request merged**